coreneuron/sim/scopmath/newton_thread.hpp

```diff
@@ -0,0 +1,183 @@
+/*
+# =============================================================================
+# Originally newton.c from SCoP library, Copyright (c) 1987-90 Duke University
+# =============================================================================
```

Comment:
same question for this file: just copy from original file I assume?

Reply:
They're not **just** copies, I made some defensive changes with symbol/function names, as now these headers will be included in many different .cpp files (including translated MOD files), so there is more scope for clashes.

For example, see https://github.com/BlueBrain/CoreNeuron/blob/8c42d359aeea351536d1135af14410d1b91b6c39/coreneuron/sim/scopmath/sparse_thread.hpp#L22-L24 and macro renaming + sanity checking https://github.com/BlueBrain/CoreNeuron/blob/8c42d359aeea351536d1135af14410d1b91b6c39/coreneuron/sim/scopmath/sparse_thread.hpp#L429-L433 and various bits of allocation/deallocation/initialisation use `new`/`delete`/constructors now that didn't before.